src/uid2DiiNormalization.ts

```diff
@@ -0,0 +1,78 @@
+enum EmailParsingState {
+  Starting,
+  SubDomain,
+}
+
+export type NormalizedPhone = string;
+export type NormalizedEmail = string;
+
+export function isNormalizedPhone(phone: string): phone is NormalizedPhone {
+  return /^\+[0-9]{10,15}$/.test(phone);
+}
+
+export function normalizeEmail(email: string): NormalizedEmail | undefined {
+  if (email == undefined || email.length <= 0) {
+    return undefined;
+  }
+
+  let preSb = "";
+  let preSbSpecialized = "";
+  let sb = "";
+  let wsBuffer = "";
+
+  let parsingState = EmailParsingState.Starting;
+  let inExtension = false;
+
+  for (const cGiven of email) {
+    let c = cGiven;
+    if (cGiven >= "A" && cGiven < "Z") {
+      c = String.fromCharCode(c.charCodeAt(0) + 32);
+    }
+
+    switch (parsingState) {
+      case EmailParsingState.Starting:
+        if (c == " ") {
```

Comment:
Is there a good reason not to use strict equality (===) checks in this function?